src/Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for setting up memory cache related services in an <see cref="IServiceCollection" />.
+    /// </summary>
+    public static class MemoryCacheServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds memory caching services to the specified <see cref="IServiceCollection" />.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        public static void AddMemoryCache(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            services.AddOptions();
+            services.TryAdd(ServiceDescriptor.Transient<IDistributedCache, LocalCache>());
```

Comment:
@Eilon Spoke to @lodejard about this. We've kept the change to AddMemoryCache and we've changed the implmentation of AddRedisCache and AddSqlServerCache to override the distributed cache in place set up by AddMemoryCache.

The rationale behind it is that AddRedisCache and AddSqlServerCache are methods that are very specifically tied up to a given implementation (Compared to AddMvc or AddDataProtection which add a bunch of abstractions) so it's ok for them to unilaterally override the distributed cache service with their implementation.

The user experience is as follows. If a user or framework needs caching it will call AddMemoryCache() which will setup the local and distributed memory caches if no cache is present.

Then if a user wants to use Redis or SQL Server as the distributed cache, they can do so by calling AddRedisCache or AddSQLServerCache.

The order in which the methods are called doesn't matter,
AddRedisCache followed by AddMemoryCache will produce the same result as AddMemoryCache folloewed by AddRedisCache.